### DISCUSSION TITLE
Error while Parallelizing a batch Simulation

### DISCUSSION THREAD

**Discussion by coprem** [ThummeTo/FMI.jl#28] [General]
Hi @ThummeTo 

I am trying to Prallelising the batch simulations.

FMU used - SpringPendulum1D

**Serial batch running Code:**
Given below is the code that I used to run the batch cases in serial. This code works perfectly fine.

```
using FMI, Plots, Distributed 
modelFMUPath = joinpath(dirname(@__FILE__), "../model/SpringPendulum1D.fmu")
FMU = fmiLoad(modelFMUPath; unpackPath = nothing)
results = []
@time for i in 0.1:0.1:1

    myFMU = fmiInstantiate!(FMU; loggingOn=false)
    fmiSetupExperiment(myFMU, 0.0)
    fmiEnterInitializationMode(myFMU)
 
    springConstant = fmiGetReal(myFMU, "spring.c") * i
    fmiSetReal(myFMU, "spring.c", springConstant)
    fmiExitInitializationMode(myFMU)

    t_start = 0.0
    t_stop = 5

    result = fmiSimulate(myFMU,t_start, t_stop; recordValues= ["mass.s"],setup=false)
    push!( results, result)

end

fmiUnload(FMU)
```

**Parallelizing Code:**
Below is the code that I tried to parallelizing the Code Above using `Distributed.jl ` but I am not able to solve it. 

```
using FMI, Plots, Distributed 
addprocs(3, exeflags=`--project=$@__DIR__`)
modelFMUPath = joinpath(dirname(@__FILE__), "../model/SpringPendulum1D.fmu")
FMU = fmiLoad(modelFMUPath; unpackPath = nothing)
results = []
@time  @distributed for i in 0.1:0.1:1

    myFMU = fmiInstantiate!(FMU; loggingOn=false)
    fmiSetupExperiment(myFMU, 0.0)
    fmiEnterInitializationMode(myFMU)
 
    springConstant = fmiGetReal(myFMU, "spring.c") * i
    fmiSetReal(myFMU, "spring.c", springConstant)
    fmiExitInitializationMode(myFMU)

    t_start = 0.0
    t_stop = 5

    result = fmiSimulate(myFMU,t_start, t_stop; recordValues= ["mass.s"],setup=false)
    push!( results, result)

end

fmiUnload(FMU)
```

can you give some suggestions where I am going wrong. I tried debugging not able to find what is the issue. 

Regards
Prem

**Comment by ThummeTo** [comment]
Dear @coprem,
we are planning support for parallelization in the next release, there will also be an example on how to use. My current guess according your issue is that the access to the DLL (or binary in general) must be handled exclusive - which it is not.

Please note: Always add the error message and error stack trace if you open an issue - the more information, the faster we can reproduce and help.

Thanks!

EDIT: I moved this topic to discussions, because this is not a library issue. I added a new feature request as issue.
